test: include tests for `responseMimeType` feature

yoichiro opened the pull request:
This Pull Request addresses the issue reported in #334, "Support for response format by specifying `responseMimeType` parameter". Specifically, it adds the new `responseMimeType` attribute to the `generationConfig`, ensuring that the `responseMimeType` attribute and its value are sent in requests to the REST API endpoint. Additionally, several test codes are added to verify that the `responseMimeType` attribute is correctly included in the requests.

yoichiro commented:
@yyyu-google Could you please review this pull request? Thank you in advance!

Agusnez commented:
Please, take a look at this PR. This is a very simple code and highly valuable! Thanks @yoichiro for addressing this!

yoichiro commented:
I created this Pull Request to add the `responseMimeType` parameter. I know that the `responseMimeType` parameter is widely used by many people. The request to add the `responseMimeType` parameter was registered as an Issue (#334), so I submitted this Pull Request to address that. I believe this is a standard procedure for contributing on GitHub.

However, the committers of this repository completely ignore Issues and Pull Requests submitted from external contributors. Not only did they ignore my Pull Request, but they also committed an almost identical addition of the `responseMimeType` parameter to the main branch. I realized this when my Pull Request conflicted with their changes.

The committers of this repository continue to ignore developers who are trying to contribute. This is very disrespectful to those developers who are making an effort to contribute. I am deeply disappointed with the committers of this repository. The fact that this repository accepts Pull Requests should mean that they welcome contributions from external sources, but unfortunately, that is not the case.

Trying to contribute to this repository is a fruitless effort. In other words, one should not attempt to contribute to this repository.

Agusnez commented:
@yoichiro It's a shame they did not recognize your work. I'm very sorry. 

You have to be proud of yourself because you did help some people, including myself. So I want to thank you personally about your disinterested work.

どうもありがとう

yinghsienwu commented:
@yoichiro Thanks for your contribution to our repo. We strongly recognize the effort and this issue has been escalated. First, your contribution well aligns with our vision, and solve some of the emergent customer request. Our team members should do a better job handling the external PR promptly, rather than what they have done. We are in the process to improve our process and SLO to review external contributions. Regarding your PR, if you desire, we may consider revert our work and merge this one. But all of these steps should be executed before our next release. In that case your contribution can be recognized. Thanks again!

yyyu-google commented:
Hi @yoichiro 
Your contribution to our repo is very valued, I apologize for not paying prompt attention to this Pull Request. I have been working on other projects when I was tagged to this Pull Request. When I created https://github.com/googleapis/nodejs-vertexai/pull/380 , I was handling another urgent case and I wasn't able to remember this Pull Request under emergency. The urgency is evident that I didn't even have unit tests included for this feature. And you are doing a great work by including test cases. I can resolve conflicts and merge your test cases in if this sounds good, however the code comment on responseMimeStyle needs to remain the same as in https://github.com/googleapis/nodejs-vertexai/pull/380 . 
Please let me know as early as your convenience if this works.

yoichiro commented:
@yinghsienwu @yyyu-google Based on your request, I have made the following changes:

Updated the comments for responseMimeType to match those in the main branch.
Changed the Pull Request title from feat to test.
I request that this Pull Request be merged into the main branch.

Additionally, yyyu-google mentioned that they "responded to an urgent customer request," but I do not believe that making a temporary fix without writing test code is the correct approach. I consider it a serious and detrimental action that compromises quality. Therefore, I firmly believe that this Pull Request represents the best code to improve the quality of this library. Moving forward, please carefully consider whether to adopt the "high-quality contributions" from external sources or the "contributions that cannot ensure quality" that you have made.

yyyu-google commented on src/models/test/models_test.ts:
```diff
@@ -280,6 +280,36 @@ describe('GenerativeModel startChat', () => {
+      '{"contents":[{"role":"user","parts":[{"text":"How are you doing today?"}]}],"generationConfig":{"candidateCount":1,"stopSequences":["hello"],"responseMimeType":"application/json"}}';
+    // @ts-ignore
+    const actualBody = fetchSpy.calls.allArgs()[0][1].body;
+    expect(actualBody).toEqual(expectedBody);
```
can you create a separate test case for this assertion? L303 and L311 are two different test intents. Same comment applies for L515.

yoichiro replied:
@yyyu-google I pushed the commit to separate the test cases at [bb069aa](https://github.com/googleapis/nodejs-vertexai/pull/364/commits/bb069aa7160f800b2b7235a80f934f750c337d38).

yoichiro commented:
@yyyu-google I updated the title of this pull request. Also, I separated the test cases you mentioned.

yyyu-google commented:
thank you @yoichiro for your contribution. i merged this Pull Request
